### PR TITLE
Fix proposal details page displays infinite loader when visiting a proposal url that doesn't exist

### PR DESCRIPTION
**Pull request by salman-neslit**
## Link to Issue
Closes: #8734

## Description of Changes
- Fixed the issue where the skeleton loader would display indefinitely for non-existent proposal URLs.
- Added a "Not Found" message when visiting a proposal URL that doesn't exist.

## Test Plan
  - Go to the proposal details page and add a wrong identifier in the URL.
  - Verified the "Not Found" message is shown correctly when visiting a non-existent proposal.

## Deployment Plan

1.

## Other Considerations

-

### CODE REVIEW

**Review by masvelio — APPROVED** [review]
added some comments, also make sure that you check your PR for automatic github warnings/errors like this 

image

File: packages/commonwealth/client/scripts/views/pages/view_proposal/index.tsx [hunk 1]
```diff
@@ -90,10 +91,14 @@ const ViewProposalPage = ({ identifier }: ViewProposalPageAttrs) => {
     }
   }, [isAdapterLoaded, proposalId]);
 
-  if (!isAdapterLoaded) {
+  if (isFetchingProposal || !isAdapterLoaded) {
     return <PageLoading message="Loading..." />;
   }
 
+  if (cosmosError) {
+    return <PageNotFound message={`We coun't find what you searched for. Try searching again`}/>;
```
[the remark this answers is not in the document]
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/pages/view_proposal/index.tsx]: Also... there's a typo here.

[the remark this answers is not in the document]
Comment (salman-neslit) [reply packages/commonwealth/client/scripts/views/pages/view_proposal/index.tsx]: Thanks for pointing that out! I've fixed the typo.

**Comment by salman-neslit** [comment]
@burtonator  @Israellund
Could you please review this PR when you get a chance?
Thanks!

**Review by Israellund — APPROVED** [review]

**Review by mzparacha — APPROVED** [review]

**Comment by masvelio** [comment]
@salman-neslit any reason why this is still not merged?

**Comment by salman-neslit** [comment]
> @salman-neslit At Common the engineers who made the PR are usually responsible for merging their PRs after they have received 2 or more reviews. Please let me know if you are unable to merge this PR and I will do it, otherwise that should be something you do going forward. Thank you.

I don't seem to have the necessary permissions to merge this pull request at the moment. Could you kindly help merge it for me this time? I’ll follow the process going forward once I have the required access.

**Comment by salman-neslit** [comment]
> @salman-neslit any reason why this is still not merged?

I don't seem to have the necessary permissions to merge this pull request at the moment. Could you kindly help merge it for me this time? I’ll follow the process going forward once I have the required access. Thank you.

**Pull request merged**